woocommerce/wc-admin
# Add store notice around processing historical data.

Fixes #1709.

This PR displays a store notice about processing historical report data. It also updates the note REST API to support filtering by status, so that we can fetch only `unactioned` store alerts (otherwise they would never get dismissed).

One thing to note is that the currently, the alert does not get dismissed unless you refresh the page.

I tried to investigate a couple ways to fix this behavior. 1) Immediately dismissing the alert when "get started" is clicked. The user didn't actually take action yet so this feels incorrect. 2) Adding a REST API update handler and updating the status of the note via JS when the process button is clicked from the settings page 3) hacking in a way to refresh the request by passing a timestamp to the notes query.

None of those options felt good to me. It seems ideally fresh-data should have a callable `refresh` method to force a fresh-data update on a certain query (allowing us to do something like like [refreshPost](https://github.com/WordPress/gutenberg/blob/c29b729a6d107fc5a7ef53efafda126d6fbb68e0/docs/designers-developers/developers/data/data-core-editor.md#refreshpost) on alerts). cc @coderkevin is this something that could be added to `fresh-data`? Any other ideas here? It seems like there may be other instances in the future where we will want to be able to requery something.

### Screenshots

<img width="969" alt="screen shot 2019-03-06 at 10 42 38 pm" src="https://user-images.githubusercontent.com/689165/53937004-c0b7c400-4079-11e9-8069-ad359a99534e.png">

### Detailed test instructions:

To Test:
* Run `phpunit` and make sure all tests pass
* Exit and restart `npm start` so you get the correct dev flags (the store alert flag was recently added)
* Deactivate WooCommerce Admin
* Delete the `wc_admin_version` option from `wp_options`, so that the activation/install hooks kick in
* Reactivate WooCommerce Admin
* Visit the dashboard and you should see the notice
